app/scripts/esrp.js

```diff
@@ -0,0 +1,238 @@
+
+function createMacNotarizeJson(pathToSign, fileName = 'test_SignInput.json') {
+  return createJson(pathToSign, [macNotarizeOp], fileName);
+}
+
+function createWinSignJson(pathToSign, fileName = 'test_SignInput.json') {
+  return createJson(pathToSign, winSignOps, fileName);
+}
+
+function createJson(pathToSign, op, fileName = 'test_SignInput.json') {
+  let rootDir = pathToSign;
+  let files = {};
+
+  // Check if we are signing one single file or all files in a directory
+  const stat = fs.statSync(pathToSign);
+  if (stat.isFile()) {
+    rootDir = path.dirname(pathToSign);
+    files = {
+      '.': [
+        {
+          path: path.basename(pathToSign),
+          hash: getSHA256(pathToSign),
+        },
+      ],
+    };
+  } else {
+    files = getFileList(pathToSign);
+  }
+
+  const filesJson = (dir, files) => {
+    return {
+      SourceLocationType: 'UNC',
+      SourceRootDirectory: path.resolve(rootDir, dir),
+      SignRequestFiles: files.map(f => ({
+        SourceLocation: f.path,
+        SourceHash: f.hash ?? '',
+        HashType: (f.hash && 'SHA256') || null,
+        Name: f.path,
+      })),
+      SigningInfo: {
+        Operations: [...op],
+      },
+    };
+  };
+
+  SIGN_JSON_TEMPLATE.SignBatches = Object.keys(files)
+    .map(dir => filesJson(dir, files[dir]))
+    .filter(f => f.SignRequestFiles.length > 0);
+  const filePath = path.join(os.tmpdir(), fileName);
+  fs.writeFileSync(filePath, JSON.stringify(SIGN_JSON_TEMPLATE, undefined, 2));
+
+  return filePath;
+}
+
+const SIGN_OPS = {
+  APPLE_SIGN: 'apple-sign',
+  APPLEN_OTARIZE: 'apple-notarize',
+  WINDOWS_SIGN: 'windows-sign',
+};
+
+/**
+ * Signs the given file, or all files in a given directory if that's what's passed to it.
+ * @param esrpTool - The path to the ESRP tool.
+ * @param op - The operation to perform. Either 'SIGN' or 'NOTARIZE'.
+ * @param pathToSign - A path to a file or directory.
+ */
+function sign(esrpTool, op, pathToSign) {
+  const absPathToSign = path.resolve(pathToSign);
+  const signJsonBase = path.basename(absPathToSign).split('.')[0];
+  let signInputJson = '';
+  if (op === SIGN_OPS.APPLE_SIGN) {
+    signInputJson = createMacSignJson(absPathToSign, `${signJsonBase}-SignInput.json`);
+  } else if (op === SIGN_OPS.APPLEN_OTARIZE) {
+    signInputJson = createMacNotarizeJson(absPathToSign, `${signJsonBase}-SignInput.json`);
+  } else if (op === SIGN_OPS.WINDOWS_SIGN) {
+    signInputJson = createWinSignJson(absPathToSign, `${signJsonBase}-SignInput.json`);
+  } else {
+    throw new Error('Invalid operation. The options are:', SIGN_OPS.join(', '));
+  }
+
+  const policyJson = path.resolve(os.tmpdir(), 'Policy.json');
+  fs.writeFileSync(policyJson, JSON.stringify(POLICY_JSON, undefined, 2));
+  const authJson = path.resolve(os.tmpdir(), 'Auth.json');
+  fs.writeFileSync(authJson, JSON.stringify(AUTH_JSON, undefined, 2));
+
+  try {
+    execSync(`${esrpTool} Sign -l Verbose -a ${authJson} -p ${policyJson} -i ${signInputJson}`);
```

Comment:
This is for building the app only and the code is not shipped in Headlamp. So I don't think there's a possibility of attack as it could be if the script was called during the execution of Headlamp.